haystack_experimental/components/extractors/llm_metadata_extractor.py

```diff
@@ -0,0 +1,220 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+
+import json
+import logging
+from typing import Any, Dict, List, Tuple, Union, Type
+from warnings import warn
+
+from haystack import Document, component, default_from_dict, default_to_dict
+from haystack.components.builders import PromptBuilder
+from haystack.components.generators import AzureOpenAIGenerator, OpenAIGenerator
+from haystack_integrations.components.generators.amazon_bedrock import AmazonBedrockGenerator
+from haystack_integrations.components.generators.google_vertex import VertexAIGeminiGenerator
```

Comment:
Should we be using Lazy imports here for the integration imports?

Reply:
This will just solve the issue about notifying the user that he/she needs to install those packages - for testing, something else needs to be added to the CI, I suspect.

Reply:
Yeah I think warning the user would still be good, but yes a different solution is needed for the CI